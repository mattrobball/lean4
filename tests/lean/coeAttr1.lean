@[coe] def f (n : Nat) : String :=
  toString n

#check f 10

instance : Coe Nat String where
  coe := f

def g (n : String) := n

#check fun x : Nat => g x
