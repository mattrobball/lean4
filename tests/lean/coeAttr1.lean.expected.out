↑10 : String
fun x => g ↑x : Nat → String
